astropy/uncertainty/function_helpers.py

```diff
@@ -62,6 +71,20 @@
 dispatched_function = FunctionAssigner(DISPATCHED_FUNCTIONS)
 
 
+def is_distribution(x):
+    from astropy.uncertainty import Distribution
+
+    return isinstance(x, Distribution)
+
+
+def get_n_samples(*arrays):
```

Comment:
I've instead added a docstring - it may well be that another override will need to check that `n_samples` is the same for all `Distribution` instances.